api/src/main/java/org/openmrs/module/queue/model/Queue.java

```diff
@@ -52,13 +55,59 @@ public class Queue extends BaseChangeableOpenmrsMetadata {
 	private Concept service;
 	
 	@OneToMany(mappedBy = "queue", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
-	@Where(clause = "voided = 0 and (started_at <= current_timestamp() and ended_at is null)")
 	private List<QueueEntry> queueEntries;
 	
 	@OneToMany(mappedBy = "queue", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
-	@Where(clause = "retired = 0")
 	private List<QueueRoom> queueRooms;
 	
+	/**
+	 * @return all non-voided QueueEntries that do not start in the future and are not already ended
+	 */
+	public List<QueueEntry> getActiveQueueEntries() {
+		List<QueueEntry> l = new ArrayList<>();
+		Date now = new Date();
+		if (queueEntries != null) {
+			for (QueueEntry e : queueEntries) {
+				if (BooleanUtils.isNotTrue(e.getVoided())) {
+					if (!e.getStartedAt().after(now) && e.getEndedAt() == null) {
+						l.add(e);
+					}
+				}
+			}
+		}
+		return l;
```

Comment:
```suggestion
	public List<QueueEntry> getActiveQueueEntries() {
		List<QueueEntry> activeQueueEntries = new ArrayList<>();
		Date now = new Date();
		if (queueEntries != null) {
			for (QueueEntry queueEntry : queueEntries) {
				if (BooleanUtils.isNotTrue(queueEntry.getVoided())) {
					if (!queueEntry.getStartedAt().after(now) && queueEntry.getEndedAt() == null) {
						activeQueueEntries.add(queueEntry);
					}
				}
			}
		}
		return activeQueueEntries;
```

Reply:
Uh... shouldn't the logic here be something like:

```java
if (!queueEntry.getStartedAt().after(now) && (queueEntry.getEndedAt() == null || queueEntry.getEndedAt().before(now)))
```